neqo-http3/src/connection.rs

```diff
@@ -335,8 +332,8 @@ impl Http3Connection {
             local_params: conn_params,
             settings_state: Http3RemoteSettingsState::NotReceived,
             streams_with_pending_data: BTreeSet::new(),
-            send_streams: HashMap::new(),
-            recv_streams: HashMap::new(),
+            send_streams: HashMap::default(),
+            recv_streams: HashMap::default(),
```

Comment:
@martinthomson do you remember why we are using `HashMap` here, but `IndexMap` in `Connection`?

Reply:
This is probably down to us paying more attention to the core transport than the HTTP/3 code.  I'd say that if we found `IndexMap` to be better at the lower layer, then it is probably every bit as good here.  The pattern of usage is nearly identical.